mmv1/products/netapp/backup.yaml

```diff
@@ -0,0 +1,145 @@
+create_verb: :POST
+update_url: projects/{{project}}/locations/{{location}}/backupVaults/{{vault_name}}/backups/{{name}}
+update_verb: :PATCH
+update_mask: true
+delete_url: projects/{{project}}/locations/{{location}}/backupVaults/{{vault_name}}/backups/{{name}}
+delete_verb: :DELETE
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+id_format: 'projects/{{project}}/locations/{{location}}/backupVaults/{{vault_name}}/backups/{{name}}'
+import_format: ['projects/{{project}}/locations/{{location}}/backupVaults/{{vault_name}}/backups/{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'netapp_backup'
+    primary_resource_id: 'test_backup'
+    vars:
+      pool_name: 'backup-pool'
+      volume_name: 'backup-volume'
+      backup_vault_name: 'backup-vault'
+      backup_name: 'test-backup'
+    test_vars_overrides:
+      network_name: 'acctest.BootstrapSharedServiceNetworkingConnection(t, "gcnv-network-config-1", acctest.ServiceNetworkWithParentService("netapp.servicenetworking.goog"))'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Location of the backup.
+  - !ruby/object:Api::Type::String
+    name: 'vault_name'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Name of the backup vault to store the backup in.
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The resource name of the backup. Needs to be unique per location.
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    description: |
+      The state of the Backup Vault.
+    values:
+      - :STATE_UNSPECIFIED
+      - :CREATING
+      - :UPLOADING
+      - :READY
+      - :DELETING
+      - :ERROR
+      - :UPDATING
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource.
+    required: false
+  - !ruby/object:Api::Type::String
+    name: 'volumeUsageBytes'
+    description: |
+      Size of the file system when the backup was created. When creating a new volume from the backup, the volume capacity will have to be at least as big.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'backupType'
+    description: |
+      Type of backup, manually created or created by a backup policy.
+    values:
+      - :TYPE_UNSPECIFIED
+      - :MANUAL
+      - :SCHEDULED
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'sourceVolume'
+    description: |
+      ID of volumes this backup belongs to. Format: `projects/{{projects_id}}/locations/{{location}}/volumes/{{name}}``
+    immutable: true
+    diff_suppress_func: tpgresource.ProjectNumberDiffSuppress
```

Comment:
Just want to double-check that all of these projectnumberdiffsuppress funcs are necessary?

Reply:
Yes we need this for 2 properties `sourceSnapshot` and `sourceVolume` as we might get project names/numbers on these

Reply:
The service uses projectNumbers internally a lot. User can input projectID or projectNumber, but will always get projectNumber returned for some fields. Without the surpress function, this will cause unnecessary execution plan diffs which will cause big issues. Unfortunately they are necessary.

Reply:
thanks for confirming!